fuseimg/data/ops/shape_ops.py

```diff
@@ -82,7 +81,214 @@ def __call__(self, sample_dict: NDict, op_id: Optional[str], key: str,
+        :param outer_height: final outer height
+        :param outer_width:  final outer width
+        :param pad_value:    value for padding around inner image
+        :number_of_channels  final number of channels in the image
+        :return:             padded image
+        """
+        inner_height, inner_width = image.shape[0], image.shape[1]
+        h_offset = int((outer_height - inner_height) / 2.0)
+        w_offset = int((outer_width - inner_width) / 2.0)
+        if self.number_of_channels > 1 :
+            outer_image = np.ones((outer_height, outer_width, self.number_of_channels), dtype=image.dtype) * pad_value
+            outer_image[h_offset:h_offset + inner_height, w_offset:w_offset + inner_width, :] = image
+        elif self.number_of_channels == 1 :
+            outer_image = np.ones((outer_height, outer_width), dtype=image.dtype) * pad_value
+            outer_image[h_offset:h_offset + inner_height, w_offset:w_offset + inner_width] = image
+        return outer_image    
+    
+    def resize_image(self,inner_image: np.ndarray, resize_to: Tuple[int,int]) -> np.ndarray :
+        """
+        resize image to the required resolution
+        :param inner_image: image of shape [H, W, C]
+        :param resize_to: required resolution [height, width]
+        :return: resized image
+        """
+        inner_image_height, inner_image_width = inner_image.shape[0], inner_image.shape[1]
+        if inner_image_height > resize_to[0]:
+            h_ratio = resize_to[0] / inner_image_height
+        else:
+            h_ratio = 1
+        if inner_image_width > resize_to[1]:
+            w_ratio = resize_to[1] / inner_image_width
+        else:
+            w_ratio = 1
+    
+        resize_ratio = min(h_ratio, w_ratio)
+        if resize_ratio != 1:
+            inner_image = skimage.transform.resize(inner_image,
+                                                   output_shape=(int(inner_image_height * resize_ratio),
+                                                                 int(inner_image_width * resize_ratio)),
+                                                   mode='reflect',
+                                                   anti_aliasing=True
+                                                   )
+        return inner_image
+    
+    def __call__(self, sample_dict: NDict, key: str , resize_to: Tuple,
+                 padding: Tuple
+        ):
+        '''
+        :param resize_to:               new size of input images, keeping proportions
+        :param padding:                 padding size
+        ''' 
+        
+        img = sample_dict[key]
+        # resize
+        if resize_to is not None:
+            img = self.resize_image(img, resize_to)
+
+        # padding
+        if padding is not None:
+            img = self.pad_image(img, padding, resize_to)
+        sample_dict[key] = img
+        return sample_dict
+        
+class OpRemoveDarkBackgroundRectangle2D(OpBase):
```

Comment:
Maybe you can explain do you mean by remove?
It crops the image, trying to remove the background in the outline?

Reply:
I renamed it, it finds the biggest non empty bbox